engine/src/main/java/org/hibernate/search/engine/backend/document/model/dsl/IndexSchemaFilterOptionsStep.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.backend.document.model.dsl;
+
+import java.util.Map;
+
+/**
+ * The final step in the definition of a filter in the index schema,
+ * where a reference to the field can be retrieved,
+ * optionally setting some parameters beforehand.
+ *
+ * @param <S> The "self" type (the actual exposed type of this step).
+ * @param <R> The reference type.
+ *
+ * @see IndexSchemaFieldFinalStep
+ */
+public interface IndexSchemaFilterOptionsStep<S extends IndexSchemaFilterOptionsStep<?, R>, R>
+	extends IndexSchemaFilterFinalStep<R> {
+
+	/**
+	 * Insert parametr to this filter definition.
+	 *
+	 * @param <T> The type of the filter parametr value.
+	 * @param name The name of the filter parametr value.
```

Comment:
The advantage of this solution is that the same "named predicate," can be used in different use cases or multiple times. It is not always required to change the parameters during the search, you can leave the default parameters unchanged. But because you can define default parameters, you can use the same "named predycate" multiple times. 

```java
public class TestAccesFactory implements NamedPredicateFactory {

    @Override
    public SearchPredicate create(NamedPredicateFactoryContext ctx) {
        SearchPredicate filter;
        String fieldPath = ctx.resolvePath("accessMode");
        String mode = (String) ctx.param("mode");

        filter = ctx.predicate()
                .match().field(fieldPath)
                .matching(mode)
                .toPredicate();

        return filter;
    }
}
```
Book entity: 
```java
@NamedPredicateBinding(name="is-readable", factory = @NamedPredicateFactoryRef(type = TestAccesFactory.class), params = {
    @NamedPredicateParam(name = "mode", value = "read")
})
@NamedPredicateBinding(name="is-writable", factory = @NamedPredicateFactoryRef(type = TestAccesFactory.class), params = {
    @NamedPredicateParam(name = "mode", value = "write")
})
public class Book {
    
    @Column
    @GenericField
    private String accessMode;
    
}
```
Document entity: 

```java
@NamedPredicateBinding(name="is-readable", factory = @NamedPredicateFactoryRef(type = TestAccesFactory.class), params = {
    @NamedPredicateParam(name = "mode", value = "can-read")
})
@NamedPredicateBinding(name="is-writable", factory = @NamedPredicateFactoryRef(type = TestAccesFactory.class), params = {
    @NamedPredicateParam(name = "mode", value = "can-write")
})
public class Document {
    
    @Column
    @GenericField
    private String accessMode;
    
}
```

```java
                BooleanPredicateClausesStep pbook = spf.bool()
                        .filter(spf.named("is-readable"));

                BooleanPredicateClausesStep pbook = spf.bool()
                        .filter(spf.named("is-writable"));
.....
                BooleanPredicateClausesStep pdocument = spf.bool()
                        .filter(spf.named("is-readable"));

                BooleanPredicateClausesStep pdocument = spf.bool()
                        .filter(spf.named("is-writable"));

                BooleanPredicateClausesStep pdocument = spf.bool()
                        .filter(spf.named("book.is-writable"));

```

As a result, the same factory can be used many times and in many use cases. Even when the entities using it are nested. Since the resulting name "named predicate" depends on the nesting location.